yabt/buildcontext.py

```diff
@@ -253,6 +255,49 @@ def done_notifier():
 
             return done_notifier
 
+        def make_retry_callback(target: Target):
+            """Return a callable "retry" notifier to
+               report a target as in need of retry.
+               Currently for tests we rebuild the target
+               when it's not necessary."""
+
+            def retry_notifier():
+                """Mark target as retry, re-entering node to end of queue"""
+                if graph_copy.has_node(target.name):
+                    ready_nodes.append(target.name)
```

Comment:
also verify that it's not already in `ready_nodes`

Reply:
If it just failed it shouldn't be in the ready_nodes, why check it?
I don't see a flow where we might push something twice to the queue.